src/proof.ts

```diff
@@ -274,32 +294,44 @@ export const relayXDomainMessages = async (
     l2TransactionHash
   )
 
+  const results: RelayResult[] = messagePairs.map((messagePair): RelayResult => {
+    return { success: relayResults.notSent, message: messagePair.message, messageProof: messagePair.proof }
+  })
   const signerWithProvider = l1Signer.connect(l1RpcProvider)
-  for (const { message, proof } of messagePairs) {
+  for (const [index, { message, proof }] of messagePairs.entries()) {
     let errorCounter = 0
+    const errors: Error[] = []
+    results[index].exceptions = errors
     while (true) {
       try {
         const result = await l1Messenger
           .connect(signerWithProvider)
           .relayMessage(message.target, message.sender, message.message, message.messageNonce, nvmTx, nvmReceipt, proof)
-        await result.wait()
-        errorCounter = 0
+        const txResponse = await result.wait()
```

Comment:
This way a consumer can start tracking a transactions life cycle independently, even if a case would arise where a user closes their browser or app prematurely to the library wrapping up the calls. It would, for now, be a reasonable solution. 

In the future we can consider splitting the message collecting and relaying. We can than leave it up to the consumer to  handle the relaying per message in a way that fits best with whatever architecture is being used.

Reply:
thanks. that was actually a bug, yea